### DISCUSSION TITLE
Unwanted permission requested by Azure AD Consent for PnP PowerShell

### DISCUSSION THREAD

**Discussion by KiranTalikoti** [pnp/powershell#508] [Q&A] [answered]
I am trying to register PnP PowerShell Enterprise app and currently app is requesting 27 different permissions .  I would  like minimize the number of permissions requested by PnP PowerShell . In my client tenant, I just have just SharePoint Admin role not the  Global Admins  or Application Admin Role. 

I had generated consent URL using the below PnP command  and shared with global admins then the request has been rejected by Azure Admin(Global Admin) since PnP requesting excess permission which is not relevant to the task I'm working.

`Register-PnPManagementShellAccess -ShowConsentUrl 
`
Is there any way to limit the permission to SharePoint and Graph API's ?

Any help would be much appreciated on this issue and thank you in advance.

**Comment by veronicageek** [comment, marked as answer]
This documentation might help you: [Authentication](https://pnp.github.io/powershell/articles/authentication.html)

You could also ask your admins to create an AzureAD App and assign the permissions individually for what you have to do. And then [connect by using your own AzureAD App](https://pnp.github.io/powershell/articles/connecting.html#connect-by-using-your-own-azure-ad-application).

**Reply by KiranTalikoti** [reply]
Thank you for Document that helped to resolve my issue.

**Reply by achimismaili** [reply]
Thanks, Veronica.
In the documentation, it says that "due to changes in the underlying SDKs", the registration is required now.

I think, it would be helpful to have an article for a compliance / IT-Sec / governance team or the M365 / SharePoint admin, why it is still a good idea and still secure to move away from the old PnP-PowerShell to the new one and approve this consent/registration.

It would be useful to have a list of a few reasons, e.g. how difficult it would be to leverage this registration for an attack by e.g. tampering/customizing the original source to a new app with same app regiastration, create an application that could act as PnP.PowerShell application in AAD or by introducing a malicious pull request? (or at least, why it is _not more critical_ than before, and so on ...)

[The old module is regarded legacy](https://docs.microsoft.com/en-us/powershell/sharepoint/sharepoint-pnp/sharepoint-pnp-cmdlets?view=sharepoint-ps), but for this new registration requirement, I did not find a good explanation anywhere, why this is now better/more secure/attractive than the one before ...

I personally see PnP.PowerShell as ready for productive use and as more or less a standard / best practice admin tool for Microsoft 365 administration. If the community has the same opinion, I think, we need to put s.th. together like a "convince your admin"- flyer / checklist. :)
